lua-api-crates/filesystem/src/lib.rs

```diff
@@ -29,6 +33,60 @@ async fn read_dir<'lua>(_: &'lua Lua, path: String) -> mlua::Result<Vec<String>>
     Ok(entries)
 }
 
+// similar (but not equal) to the shell command basename
+async fn basename<'lua>(_: &'lua Lua, path: String) -> mlua::Result<String> {
```

Comment:
These path-name-only operations don't need to be async; it's a bit more efficient not to use async functions in the lua runtime, so it is desirable to avoid it if we don't need it!

```suggestion
fn basename<'lua>(_: &'lua Lua, path: String) -> mlua::Result<String> {
```